doc/user-guide/user-guide.rst

```diff
@@ -122,107 +126,262 @@ To deal with this process in a more comprehensive way, this repository uses a di
+
+      Eventually this will be enforced by the build system.
+
+  * Update the version of all required packages in ``cycle/cycle.env``.
+
+  * Commit and push your changes.
+
+    .. prompt:: bash
+
+      git push
+
+  * It is important to follow the build steps order.
+    Also, we recommend running one step at a time in the Jenkins server, to make sure the image is pushed correctly, avoiding a potential push problem at the end.
+
+    The build should always start with the base images; ``build_conda_package_builder``, ``deploy_conda`` and ``deploy_lsstsqre``.
+    First build the ``build_conda_package_builder`` images.
+    Since ``deploy_conda`` is quicker to build it is, in general, preferable to continue with that one.
+
+    .. note::
+
+      The most common issue in building the base images is that the core conda packages are not built yet.
+      Most notably, the ``ts-idl`` package.
+
+      It is, in general, required to run the build manually by going to the `ts-idl conda package job`_ in Jenkins, selecting the version of ``ts-idl`` to build in the "Tags" tab and building it with the appropriate version of ``ts-xml`` and ``ts-sal``.
+      Note that this build may also fail if the RPM artifacts are missing.
+      In this case, contact the build team in the `ts-build slack channel`_.
+
+    .. _ts-idl conda package job: https://tssw-ci.lsst.org/view/Conda_Jobs/job/IDL_Conda_package/
+    .. _ts-build slack channel: https://lsstc.slack.com/archives/CM6NYMG3F
+
+    Once both base images are built the system is ready to build the remaining components.
+    The first thing to do is to build the ts-cycle conda package.
+    This package depends on the core packages and can be built at any time so now is as good a time as any.
+    Next, given the simplicity and overall time it takes to complete, it is advisable to build the ``base_components`` next.
+    This step will build the majority of the systems.
+    Specifically, those that use the ``deploy_conda`` and are built from conda packages.
+
+    .. note::
+
+      One of the most common issues in building this step is when the selected version of the conda package for a component (or library) is not available.
+      In this case, make sure to check the conda Jenkins build for the particular package.
+      On some occasions developers release the code before all required dependencies are available (in most cases the idl package) and the release build fails.
+      For most cases, rerunning the build in Jenkins (after making sure the dependencies are available) should be enough to fix the problem.
+      If further issues are encountered with the build for that particular package you can either attempt to fix it yourself (most cases are simple pep8 or black formatting issues) or contact the developer in charge of the component and request a patch.
```

Comment:
Michael created the vanward project (https://github.com/lsst-ts/vanward) with scripts that check and make sure that all packages are ready to be build. The only thing it doesn't do (yet?) is to check the version history of each package to verify that a tag with the latest version has been created but otherwise it does a very good job.